src/Features/Core/Portable/ReassignedVariable/AbstractReassignedVariableService.cs

```diff
@@ -0,0 +1,326 @@
+            {
+                var local = semanticModel.GetDeclaredSymbol(designation, cancellationToken) as ILocalSymbol;
+                if (IsSymbolReassigned(local))
+                    result.Add(GetIdentifierOfSingleVariableDesignation(designation).Span);
+            }
+
+            bool IsSymbolReassigned([NotNullWhen(true)] ISymbol? symbol)
+            {
+                // Note: we don't need to test range variables, as they are never reassignable.
+                if (symbol is not IParameterSymbol and not ILocalSymbol)
+                    return false;
+
+                if (!symbolToIsReassigned.TryGetValue(symbol, out var reassignedResult))
+                {
+                    reassignedResult = symbol is IParameterSymbol parameter
+                        ? ComputeParameterIsAssigned(parameter)
+                        : ComputeLocalIsAssigned((ILocalSymbol)symbol);
+                    symbolToIsReassigned[symbol] = reassignedResult;
+                }
+
+                return reassignedResult;
+            }
+
+            bool ComputeParameterIsAssigned(IParameterSymbol parameter)
+            {
+                if (!TryGetParameterLocation(parameter, out var parameterLocation))
+                    return false;
+
+                var methodOrProperty = parameter.ContainingSymbol;
+
+                // If we're on an accessor parameter. Map up to the matching parameter for the property/indexer.
+                if (methodOrProperty is IMethodSymbol { MethodKind: MethodKind.PropertyGet or MethodKind.PropertySet } method)
+                    methodOrProperty = method.AssociatedSymbol as IPropertySymbol;
+
+                if (methodOrProperty is not IMethodSymbol and not IPropertySymbol)
+                    return false;
+
+                if (methodOrProperty.DeclaringSyntaxReferences.Length == 0)
+                    return false;
+
+                // Potentially a reference to a parameter in another file.
+                var methodOrPropertyDeclaration = methodOrProperty.DeclaringSyntaxReferences.First().GetSyntax(cancellationToken);
+                if (methodOrPropertyDeclaration.SyntaxTree != semanticModel.SyntaxTree)
+                    return false;
+
+                // All parameters (except for 'out' parameters), come in definitely assigned.
+                return AnalyzePotentialMatches(
+                    parameter,
+                    parameterLocation,
+                    symbolIsDefinitelyAssigned: parameter.RefKind != RefKind.Out,
+                    GetMemberBlock(methodOrPropertyDeclaration));
+            }
+
+            bool TryGetParameterLocation(IParameterSymbol parameter, out TextSpan location)
+            {
+                if (parameter.Locations.Length > 0)
+                {
+                    var parameterLocation = parameter.Locations[0];
+                    if (parameterLocation.SourceTree == semanticModel.SyntaxTree)
```

Comment:
Why this check is needed?

Reply:
it's paranoia for things like named params, or partial method params.  i can make into a contrat call though .